test/compress/functions.js

```diff
@@ -5283,3 +5283,98 @@ issue_4471: {
         "PASS",
     ]
 }
+
+issue_4612_1: {
+    options = {
+        inline: true,
+        reduce_vars: true,
+        unused: true,
+    }
+    input: {
+        (function() {
+            function f() {
+                return g();
+            }
+            function g() {
+                return f();
+            }
+            g();
+        })();
+    }
+    expect: {
+        (function() {
+            function f() {
+                return f();
+            }
+            f();
+        })();
+    }
+    expect_stdout: RangeError("Maximum call stack size exceeded")
```

Comment:
You don't see that every day in a test - trying to blow the call stack.
I take it that any attempt to limit the recursion depth prevents the bug from being triggered?

Reply:
This test case exercises the bug and completes normally:
```
(function(x) {
    function f() {
        return g();
    }
    function g(z) {
        return z || f();
    }
    console.log(g(x));
})("PASS");
```

Reply:
Cool &minus; I tried to mutate the return statement in `f()` and gave up too early 😅